boot/zephyr/tinycbor/include/tinycbor/assert_p.h

```diff
@@ -0,0 +1,29 @@
+/****************************************************************************
+**
+** Copyright (C) 2015 Intel Corporation
+**
+** Permission is hereby granted, free of charge, to any person obtaining a copy
+** of this software and associated documentation files (the "Software"), to deal
+** in the Software without restriction, including without limitation the rights
+** to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+** copies of the Software, and to permit persons to whom the Software is
+** furnished to do so, subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be included in
+** all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+** OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+** THE SOFTWARE.
+**
+****************************************************************************/
+
```

Comment:
This comes from mainstream tinycbor. If we want to add header guards we should PR the main project first instead of making even more modifications than we already did. We should strive to be in sync...

Reply:
Whole tinycbor library here ia temporary hack (see carles comment above https://github.com/runtimeco/mcuboot/pull/152#issuecomment-342554984). Threa are already works in progress for provide `good enaught` unstreamed tinycbor for zephyr, but we won't to wait for it. I would keep this library as is because it doesn't make any value to update it. I can add guards in mcuboot tinycbor copy if you wan (?)

Reply:
>  I can add guards in mcuboot tinycbor copy if you wan (?)
No, that's not necessary then, was not aware that this was forked code.